### PR TITLE
Add workshops to catalog UI

### PR DESCRIPTION
**Pull request by jkupferer**

### CODE REVIEW

File: catalog/ui/src/app/Catalog/Catalog.tsx [hunk 1]
```diff
@@ -294,14 +296,23 @@ const Catalog: React.FunctionComponent = () => {
...
-        />
-      );
+      if (showRequestForm) {
+        return (
+          <CatalogItemRequestForm
+            catalogItem={openCatalogItem}
+            onCancel={onRequestCancel}
+          />
+        );
+      } else {
```
Comment (aleixhub) [inline catalog/ui/src/app/Catalog/Catalog.tsx:308]: this line is could be removed

Comment (jkupferer) [reply catalog/ui/src/app/Catalog/Catalog.tsx:308]: Can you explain?

Comment (aleixhub) [reply catalog/ui/src/app/Catalog/Catalog.tsx:308]: Sure, 
Sorry i meant the else condition could be removed, I wasnt to clear in my previous comment.
So there is a return inside the if, which means it will stop and exit this context. Therefore, if the first condition is not applied you want to execute the else (which doesn't matter if is inside the else statement or not, because will be executed regardless).

In code these tho blocks does the same:

```
function() {
  if (condition) {
    return something;
  } else {
    return somethingElse;
   }
}
```

```
function() {
  if (condition) {
    return something;
  }
  return somethingElse;
}
```

File: catalog/ui/src/app/Catalog/CatalogItemDetails.tsx [hunk 2]
```diff
@@ -127,6 +130,11 @@ const CatalogItemDetails: React.FunctionComponent<CatalogItemDetailsProps> = ({
     }
   }
 
+  async function requestWorkshop(): Promise<void> {
```
Comment (aleixhub) [inline catalog/ui/src/app/Catalog/CatalogItemDetails.tsx:133]: I think this could be a regular function (without the async and the Promise)

File: catalog/ui/src/app/Catalog/CatalogItemWorkshopForm.tsx [hunk 3]
```diff
@@ -0,0 +1,628 @@
...
+}
+
+function cancelFormStateConditionChecks(state:FormState): void {
+  if (state) {
+    state.conditionChecks.canceled = true;
+  }
+}
+
+// Because salesforce checks are asynchronous they need to be resolved before checking the condition logic
+async function _checkCondition(condition: string, vars: ConditionValues): Promise<boolean> {
```
Comment (aleixhub) [inline catalog/ui/src/app/Catalog/CatalogItemWorkshopForm.tsx:110]: checkSalesforceCondition could be a better naming?

File: catalog/ui/src/app/Catalog/CatalogItemWorkshopForm.tsx [hunk 4]
```diff
@@ -0,0 +1,628 @@
...
+      }
+      return () => {
+        if (componentWillUnmount.current) {
+          cancelFormStateConditionChecks(formState);
+        }
+      }
+    } else {
+      return null;
+    }
+  }, [formState]);
```
Comment (aleixhub) [inline catalog/ui/src/app/Catalog/CatalogItemWorkshopForm.tsx:418]: im not sure this will detect object changes as expected, will be better to send directly the [formState.conditionChecks.complete] or use a deep compare strategy (maybe if the object is shallow enough convert the object to string)

File: catalog/ui/src/app/Workshops/AllWorkshops.tsx [hunk 5]
```diff
@@ -0,0 +1 @@
+import React from "react";
```
Comment (aleixhub) [inline catalog/ui/src/app/Workshops/AllWorkshops.tsx:1]: is this file needed?

File: catalog/ui/src/app/Workshops/workshops.css [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+.workshops-item-body.pf-c-page__main-section {
+  padding-top: 0px;
+}
+
+.workshops-item-body.pf-c-page__main-section .pf-c-tabs {
+  margin-left: -100px;
+  padding-left: 100px;
+  margin-right: -100px;
+  padding-right: 100px;
+  margin-bottom: 10pt;
```
Comment (aleixhub) [inline catalog/ui/src/app/Workshops/workshops.css:21]: this isnt important for this PR, so you can ignore it for now.
but there is a reason to use pt?

File: catalog/ui/src/app/components/EditableText.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        <TextArea autoFocus
+          aria-label={ariaLabel}
+          className="editable-text-area"
+          onBlur={() => finishEditing()}
+          onChange={(v) => setEditedValue(v)}
+          onKeyUp={onKeyUp}
+          value={editedValue}
+        />
+      );
+    } else {
```
Comment (aleixhub) [inline catalog/ui/src/app/components/EditableText.tsx:74]: same as before, this else statement could be removed

File: catalog/ui/src/app/components/EditableText.tsx [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+          aria-label={ariaLabel}
+          className="editable-text-input"
+          onBlur={() => finishEditing()}
+          onChange={(v) => setEditedValue(v)}
+          onKeyUp={onKeyUp}
+          value={editedValue}
+        />
+      );
+    }
+  } else {
```
Comment (aleixhub) [inline catalog/ui/src/app/components/EditableText.tsx:86]: same as before, this else statement could be removed

File: catalog/ui/src/app/components/EditableText.tsx [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+        <Text
+          component={TextVariants.p}
+          className={value ? 'editable-text-value' : 'editable-text-placeholder'}
+          onClick={beginEditing}
+        >
+          {value ? '********' : '- none -'}
+          <Button onClick={beginEditing} variant="link" icon={updating ? <Spinner size="sm"/> : <PencilAltIcon/>}/>
+       </Text>
+      )
+    } else {
```
Comment (aleixhub) [inline catalog/ui/src/app/components/EditableText.tsx:98]: same as before, this else statement could be removed

File: catalog/ui/src/app/Catalog/Catalog.tsx (hunk 1 above)
Comment (jkupferer) [reply catalog/ui/src/app/Catalog/Catalog.tsx:308]: I like it. I'll change it.

File: catalog/ui/src/app/Catalog/CatalogItemDetails.tsx (hunk 2 above)
Comment (jkupferer) [reply catalog/ui/src/app/Catalog/CatalogItemDetails.tsx:133]: Good catch, thanks!

File: catalog/ui/src/app/Catalog/CatalogItemWorkshopForm.tsx (hunk 3 above)
Comment (jkupferer) [reply catalog/ui/src/app/Catalog/CatalogItemWorkshopForm.tsx:110]: It is checking more than just the salesforce value. It just just that the saleforce check requires an asynchronous call.

File: catalog/ui/src/app/Catalog/CatalogItemWorkshopForm.tsx (hunk 4 above)
Comment (jkupferer) [reply catalog/ui/src/app/Catalog/CatalogItemWorkshopForm.tsx:418]: The `formState` is a new object each time because of how the reducer works. I am not happy with this code, but it does work as expected. I think it could be done better. Also there is duplication here and the `CatalogItemRequestForm.tsx` that should be cleaned up, but I think that is work for a later PR.

File: catalog/ui/src/app/Workshops/AllWorkshops.tsx (hunk 5 above)
Comment (jkupferer) [reply catalog/ui/src/app/Workshops/AllWorkshops.tsx:1]: Thanks! Removing

File: catalog/ui/src/app/Workshops/workshops.css (hunk 6 above)
Comment (jkupferer) [reply catalog/ui/src/app/Workshops/workshops.css:21]: No good reason. There is also duplication between this and `ui/src/app/Services/services.css` that should get cleaned up. I think a lot of the styles need attention and rework.

File: catalog/ui/src/app/components/EditableText.tsx (hunk 7 above)
Comment (jkupferer) [reply catalog/ui/src/app/components/EditableText.tsx:74]: fixed.

File: catalog/ui/src/app/components/EditableText.tsx (hunk 8 above)
Comment (jkupferer) [reply catalog/ui/src/app/components/EditableText.tsx:86]: fixed

File: catalog/ui/src/app/components/EditableText.tsx (hunk 9 above)
Comment (jkupferer) [reply catalog/ui/src/app/components/EditableText.tsx:98]: fixed

File: catalog/ui/src/app/Catalog/CatalogItemWorkshopForm.tsx (hunk 4 above)
Comment (aleixhub) [reply catalog/ui/src/app/Catalog/CatalogItemWorkshopForm.tsx:418]: Makes sense, is ok like this then 👌🏼

File: catalog/ui/src/app/Workshops/workshops.css (hunk 6 above)
Comment (aleixhub) [reply catalog/ui/src/app/Workshops/workshops.css:21]: Agree, we can work on this later

**Review by aleixhub — APPROVED** [review]
LGTM 🙌
